locmap/lib/deletion.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+/*
+    Deleting *all* data related to a user from the database.
+*/
+
+var db = require('../../lib/db');
+var logger = require('../../lib/logger');
+var conf = require('../../lib/config');
+var LocMapUserModel = require('./locMapUserModel');
+var LocMapShareModel = require('./locMapShareModel');
+
+var assert = require('assert');
+var suspend = require('suspend');
+
+var userPrefix = conf.get('db').userPrefix;
+var sharePrefix = conf.get('db').sharePrefix;
+var codePrefix = conf.get('db').deleteCodePrefix; 
+
+exports.deleteUser = suspend(function* (userId, callback) {
+
+    try {
+        yield db.del(userPrefix + userId, suspend.resume());
+        yield db.del(sharePrefix + userId, suspend.resume());
+        yield db.del(codePrefix + userId, suspend.resume());
+
+        var shareKeys = yield db.keys(sharePrefix+'*', suspend.resume());
+
+        for (var i = 0; i < shareKeys.length; ++i) {
+
+            var id = shareKeys[i].split(':')[1];
+            var share = new LocMapShareModel(id);
+
+            var result = yield share.getData(suspend.resumeRaw());
+            
+            if (result[0] === 404) {
```

Comment:
Are you SURE result is an array? Coult it be undefined in any situation? If you know for sure it is always defined, then this is fine.